src/pages/publication/ReferencesPanel.tsx

```diff
@@ -34,8 +34,8 @@ const ReferencesPanel: FC<PanelProps> = ({ setTouchedFields }) => {
 
   useEffect(
     // Set all fields as touched if user navigates away from this panel (on unmount)
-    () => () => setTouchedFields(touchedReferenceTabFields),
-    [setTouchedFields]
+    () => () => setTouchedFields(touchedReferenceTabFields(publicationContextType)),
+    [setTouchedFields, publicationContextType]
```

Comment:
har ikke heeelt kontroll på dette, men vil ikke denne cleanup-funksjonen kjøre når man bytter type også nå når publicationContextType er en dependency? altså hvis man endrer fra Journal til Book vil touched i formik si at alle Journal-felter er touched og dermed vise feilmeldinger der?

Reply:
jo, det er sant. Hva gjør denne egentlig? Når jeg utkommenterer hele useEffecten, så vil feltene fortsatt settes til touched når man navigerer bort. Er det duplikat-kode? Ser ut som jeg bare kan fjerne det? 🤔

Reply:
Den useEffecten i SubmissionPanel tar vel av seg dette (setter felt til touched på mount)? Så den trengs ikke?

Reply:
når du går til SubmissionPanel settes alle felter i alle tabs til touched, men vi trenger vel denne til å sette Reference-felter til touched når man feks navigerer fra reference til files

Reply:
ahhh, selvfølgelig. hmmm... ja, jeg får finne ut av hvordan jeg kan løse det da 🤔

Reply:
aha, hadde selvsagt akkurat samme problemet for files og contributors, så du kan se der! måtte ta in useRef

Reply:
ja, fant ut av det rett etter jeg skrev kommentaren der 😊